sentry/src/routes/channel.rs

```diff
@@ -342,6 +336,82 @@ pub async fn get_spender_limits<A: Adapter + 'static>(
     Ok(success_response(serde_json::to_string(&res)?))
 }
 
+pub async fn get_all_spender_limits<A: Adapter + 'static>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let channel = req
+        .extensions()
+        .get::<ChannelV5>()
+        .expect("Request should have Channel")
+        .to_owned();
+
+    let new_state = get_corresponding_new_state(&app.pool, &channel)
+        .await
+        .ok_or(ResponseError::NotFound)?;
+
+    let mut all_spender_limits: HashMap<Address, Spender> = HashMap::new();
+
+    // Using for loop to avoid async closures
+    for (spender_addr, balance) in new_state.msg.balances.spenders.iter() {
+        let latest_spendable =
+            match fetch_spendable(app.pool.clone(), spender_addr, &channel.id()).await? {
+                Some(spendable) => spendable,
+                None => return Err(ResponseError::NotFound),
+            };
+
+        let total_deposited = latest_spendable.deposit.total;
+        let spender_leaf = get_spender_leaf_for_spender(balance, &total_deposited);
+
+        let spender_info = Spender {
+            total_deposited,
+            spender_leaf,
+        };
+
+        all_spender_limits.insert(*spender_addr, spender_info);
+    }
+
+    let res = AllSpendersResponse {
+        spenders: all_spender_limits,
+    };
+
+    Ok(success_response(serde_json::to_string(&res)?))
+}
+
+async fn get_corresponding_new_state(
+    pool: &DbPool,
+    channel: &ChannelV5,
+) -> Option<MessageResponse<NewState<UncheckedState>>> {
+    let approve_state = match latest_approve_state_v5(pool, channel).await.ok()? {
+        Some(approve_state) => approve_state,
+        None => return None,
+    };
+
+    let state_root = approve_state.msg.state_root.clone();
+
+    let new_state = latest_new_state_v5(pool, channel, &state_root).await.ok()?;
+
+    new_state
+}
+
+fn get_spender_leaf_for_spender(
+    spender_balance: &UnifiedNum,
+    total_deposited: &UnifiedNum,
+) -> Option<SpenderLeaf> {
+    let total_spent = match total_deposited.checked_sub(spender_balance) {
+        Some(spent) => spent,
+        None => return None,
+    };
```

Comment:
If I remember correctly , you can use `?` on an `Option` when your function returns an `Option`, the same way as `Result`.

```suggestion
    let total_spent = match total_deposited.checked_sub(spender_balance)?;
```